plugin/core/registry.py

```diff
@@ -144,11 +143,33 @@ def client_with_capability(self, capability: str) -> Optional[Client]:
         return client_from_session(session_for_view(self.view, capability))
 
 
-class LspRestartClientCommand(sublime_plugin.TextCommand):
+class LspRestartAllServersCommand(sublime_plugin.TextCommand):
     def is_enabled(self) -> bool:
         return is_supported_view(self.view)
 
     def run(self, edit: Any) -> None:
         window = self.view.window()
         if window:
             windows.lookup(window).restart_sessions()
+
+
+class LspRestartServerCommand(sublime_plugin.TextCommand):
+    def is_enabled(self) -> bool:
+        window = self.view.window()
+        lsp_servers = bool()
+        if window:
+            lsp_servers = True if windows.lookup(window).get_sessions() else False
+        return is_supported_view(self.view) and lsp_servers
```

Comment:
Was also thinking of creating a function in the WindowManager as such

```python
def has_active_sessions(self) -> bool:
    return True if list(self._sessions) else False
```

and updating this code to look like

```python
class LspRestartServerCommand(sublime_plugin.TextCommand):
    def is_enabled(self) -> bool:
        window = self.view.window()
        if not window:
            return False
        return is_supported_view(self.view) and windows.lookup(window).has_active_sessions()
```

Reply:
Thoughts